### PR TITLE
Framework for automation

### PR DESCRIPTION
**Pull request by AnastasiaGrinchik**

### CODE REVIEW

File: bisinessObject/Calculator.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+export class Calculator {
+    constructor(calculatorParametrs) {
+        this.numberOfInstance = calculatorParametrs.numberOfInstance;
+        this.operatingSystem = calculatorParametrs.operatingSystem;
+        this.machineClass = calculatorParametrs.machineClass;
+        this.series = calculatorParametrs.series;
+        this.machineType = calculatorParametrs.machineType;
+        this.gruType = calculatorParametrs.gruType;
```
Comment (rimus) [inline bisinessObject/Calculator.js]: Typo: gpu.

File: bisinessObject/Calculator.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+        this.numberOfGrus = calculatorParametrs.numberOfGrus;
```
Comment (rimus) [inline bisinessObject/Calculator.js]: Typo: gpu.

File: bisinessObject/CalculatorFactory.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+export class CalculatorFactory {
+    static getCalculatorForm() {
+        let calculatorParametrs = {
+            numberOfInstance: '4',
+            operatingSystem:
+                'Free: Debian, CentOS, CoreOS, Ubuntu or BYOL (Bring Your Own License)',
+            machineClass: 'Regular',
+            series: 'N1',
+            machineType: 'n1-standard-8 (vCPUs: 8, RAM: 30GB)',
+            gruType: 'NVIDIA Tesla V100',
```
Comment (rimus) [inline bisinessObject/CalculatorFactory.js]: Typo: gpu.

File: bisinessObject/CalculatorFactory.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+            numberOfGrus: '1',
```
Comment (rimus) [inline bisinessObject/CalculatorFactory.js]: Typo: gpu.

File: bisinessObject/PasteFactory.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+            name: 'how to gain dominance among developers',
+            text: 'git config --global user.name "New Sheriff in Town"\ngit reset $(git commit-tree HEAD^{tree} -m "Legacy code")\ngit push origin master --force',
+            expirationTime: '10 Minutes',
+            syntax: 'Bash',
+        };
+        return new Paste(pasteParametrs);
+    }
+}
+// let FirstPaste = PasteFactory.getFirstPaste();
+// console.log(FirstPaste);
```
Comment (rimus) [inline bisinessObject/PasteFactory.js]: Clear please.

File: pageObject_model/CalculatorHomePage.js [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+        this.machineTypeListXpath = '//*[@id="select_container_103"]';
+        this.checkboxXpath =
+            '//*[@aria-label="Add GPUs"]/div[@class="md-container md-ink-ripple"]';
+        this.numberGrusSelectXpath =
+            '//*[@placeholder="Number of GPUs"]/*[@class="md-select-value"]//div';
+        this.numberGrusListXpath = '//*[@id="select_container_454"]';
+        this.numberGrusItemXpath =
+            '//*[@id="select_option_462"]/div[@class="md-text ng-binding"]';
+        this.GruTypeSelectXpath = '//*[@aria-label="GPU type"]';
+        this.GruTypeListXpath = '//*[@value="NVIDIA_TESLA_K80"]/ancestor::div';
```
Comment (rimus) [inline pageObject_model/CalculatorHomePage.js]: 1) Typo: gpu
2) Only classes should start with capitals.

File: wdio.conf.js [hunk 7]
```diff
@@ -0,0 +1,337 @@
...
+        test,
+        context,
+        { error, result, duration, passed, retries }
+    ) {
+        function getTitleForScreenshot() {
+            let year = new Date().getFullYear();
+            let mounth = new Date().getMonth();
+            let day = new Date().getDate();
+            let hour = new Date().getHours();
+            let minutes = new Date().getMinutes();
```
Comment (rimus) [inline wdio.conf.js]: Make before this:
`let date = new Date();`
and then
`let year = date.getFullYear();` etc.

P.S. `getMonth()` ? :)
P.P.S. Typo: month (line 274)

File: wdio.conf.js [hunk 8]
```diff
@@ -0,0 +1,337 @@
...
+            return `${year}_${mounth}_${day}-${hour}_${minutes}`;
```
Comment (rimus) [inline wdio.conf.js]: Maybe add seconds to filename?

**Review by rimus — CHANGES_REQUESTED** [review]

File: bisinessObject/Calculator.js (hunk 1 above)
Comment (AnastasiaGrinchik) [reply bisinessObject/Calculator.js]: done

File: bisinessObject/Calculator.js (hunk 2 above)
Comment (AnastasiaGrinchik) [reply bisinessObject/Calculator.js]: done

File: bisinessObject/CalculatorFactory.js (hunk 3 above)
Comment (AnastasiaGrinchik) [reply bisinessObject/CalculatorFactory.js]: done

File: bisinessObject/CalculatorFactory.js (hunk 4 above)
Comment (AnastasiaGrinchik) [reply bisinessObject/CalculatorFactory.js]: done

File: bisinessObject/PasteFactory.js (hunk 5 above)
Comment (AnastasiaGrinchik) [reply bisinessObject/PasteFactory.js]: cleared

File: pageObject_model/CalculatorHomePage.js (hunk 6 above)
Comment (AnastasiaGrinchik) [reply pageObject_model/CalculatorHomePage.js]: done

File: wdio.conf.js (hunk 7 above)
Comment (AnastasiaGrinchik) [reply wdio.conf.js]: fixed

File: wdio.conf.js (hunk 8 above)
Comment (AnastasiaGrinchik) [reply wdio.conf.js]: added seconds

**Pull request merged**